DotNetDBF/DBFReader.cs

```diff
@@ -245,175 +244,156 @@ internal Object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
+                                        int.TryParse(monthString, out month) &&
+                                        int.TryParse(dayString, out day))
+                                    {
+                                        recordObjects[i] = new DateTime(year, month, day);
+                                    }
+                                    else
+                                    {
+                                        recordObjects[i] = null;
+                                    }
                                 }
-                                else
+                                catch (ArgumentOutOfRangeException)
                                 {
+                                    /* this field may be empty or may have improper value set */
                                     recordObjects[i] = null;
                                 }
-                            }
-                            catch (ArgumentOutOfRangeException)
-                            {
-                                /* this field may be empty or may have improper value set */
-                                recordObjects[i] = null;
-                            }
-
-                            break;
 
+                                break;
+                            }
                         case NativeDbType.Float:
-
-                            try
                             {
-                                byte[] t_float = new byte[
-                                    _header.FieldArray[i].FieldLength
-                                    ];
-                                _dataInputStream.Read(t_float, 0, t_float.Length);
-                                String tParsed = CharEncoding.GetString(t_float);
-                                var tLast = tParsed.Substring(tParsed.Length - 1);
-                                if (tParsed.Length > 0
-                                    && tLast != " "
-                                    && tLast != DBFFieldType.Unknown)
+                                try
                                 {
-                                    recordObjects[i] = Double.Parse(tParsed, NumberStyles.Float | NumberStyles.AllowLeadingWhite);
+                                    byte[] t_float = new byte[_header.FieldArray[i].FieldLength];
+                                    _dataInputStream.Read(t_float, 0, t_float.Length);
+                                    string tParsed = CharEncoding.GetString(t_float);
+                                    string tLast = tParsed.Substring(tParsed.Length - 1);
+                                    if (tParsed.Length > 0
+                                        && tLast != " "
+                                        && tLast != DBFFieldType.Unknown)
+                                    {
+                                        recordObjects[i] = Convert.ToSingle(tParsed, NumberFormatProvider);

```

Comment:
As far as I can find info on the DBF format, it can't handle anything but Numbers and Periods and minus signs in the data, so you can't use any other format other than NumberFormatInfo.InvariantInfo to read that data, so it should just be hardcoded like it is for writing. The bug you are fixing is that it was using the Current Culture Number Format for reading which worked for everyone whose culture use periods for decimal points but was not accurate.

 Really though this should just be `Double.Parse(tParsed, NumberStyles.AllowLeadingWhite |  NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint, NumberFormatProvider.InvariantInfo)` 

I didn't notice that you prior that you switched from double precision to single precision floating points, I'm not sure what's right, because the DBF format is actually storing it like a decimal, even though it labels it as a float, and the DBF also has a double type that even stores as a double that I never implement. I probably did this based off the Java Library I ported from. If we just want to fix this bug and not cause people problems who have been using this for years just to fix this bug and keep it a double.